lib/api/errors/error_base.rb

```diff
@@ -30,33 +30,64 @@
 module API
   module Errors
     class ErrorBase < Grape::Exceptions::Base
-      attr_reader :code, :message, :details, :errors
+      attr_reader :code, :message, :details, :errors, :property
 
-      def self.create(errors)
-        [:error_not_found, :error_unauthorized, :error_conflict, :error_readonly].each do |key|
-          if errors.has_key?(key)
-            case key
-            when :error_not_found
-              return ::API::Errors::NotFound.new(errors[key].join(' '))
-            when :error_unauthorized
-              return ::API::Errors::Unauthorized
-            when :error_conflict
-              return ::API::Errors::Conflict
-            when :error_readonly
-              return ::API::Errors::UnwritableProperty.new(errors[key].flatten)
+      class << self
+        ##
+        # Converts the given ActiveRecord errors into an Array of Error objects
+        # (i.e. subclasses of ErrorBase)
+        # In case the given errors contain 'critical' errors, the resulting Array will only
+        # contain the critical error and no non-critical errors (avoiding information disclosure)
+        # That means: The returned errors are always safe for display towards a user
+        def create_errors(errors)
```

Comment:
Is the number of these critical errors likely to grow in the foreseeable future? The refactor here is fine for three errors, but I can see this getting out of hand If you add a new error here.

**It is certainly out of scope for this PR, being as large as it is**, but when new errors are added, we could probably extract this method and simply iterate over all possibly errors to let them check if they apply.